lib/avo/base_resource.rb

```diff
@@ -295,10 +295,17 @@ def available_view_types
       view_types = [:table]
 
       view_types << :grid if get_grid_fields.present?
+      view_types << :map if mappable?
 
       view_types
     end
 
+    def mappable?
```

Comment:
There is a slight mismatch between this method and the `resource_location_markers` helper method above. This one returns true if the resource has _any_ location fields, while the helper method only returns markers if the resource has a `coordinates` method.

I have not thought much about allowing the developer to specify a location field name other than `coordinates` for the map. It seems like the GridCollector provides a pattern for doing that sort of thing.

Reply:
I resolved this by making the method look for the field by both type and name, sot hat it matches the helper method. If it's not correct, it is at least consistent.